inc/Engine/Preload/Controller/PreloadUrl.php

```diff
@@ -59,6 +61,16 @@ public function __construct( Options_Data $options, Queue $queue, Cache $rocket_
 	 */
 	public function preload_url( string $url ) {
 
+		/**
+		 * Filters to allow query string in preload.
+		 *
+		 * @param array $is_allowed Are query strings allowed.
+		 */
+		if ( apply_filters( 'rocket_preload_query_string', false ) && ! $this->has_cached_query_string( $url ) ) {
```

Comment:
```suggestion
		if ( ! $this->has_cached_query_string( $url ) && apply_filters( 'rocket_preload_query_string', false ) ) {
```
Reverse the check, dealing with string is faster than applying the filter so if we didn't find any query string, we don't need to check the filter at all.
(Small performance enhancement)